### ISSUE TITLE
Circular Dependency Error After Removing From Main Source Set

### ISSUE THREAD

**Issue by aorze** [unbroken-dome/gradle-xjc-plugin#11]
I am attempting to separate my xjc generated sources from the main source set (for project specific reasons).  However, it keeps failing due to a circular dependency error.

My build.gradle contains:

```
xjc {
  includeInMainCompilation = false
}

sourceSets {
    generated { java { srcDir xjcGenerate.outputDirectory } }
}

compileGeneratedJava {
    dependsOn xjcGenerate
}

dependencies {
    compile sourceSets.generated.output
}
```

The actual error message is:
```
* What went wrong:
Circular dependency between the following tasks:
:compileGeneratedJava
\--- :xjcGenerate
     \--- :generatedClasses
          \--- :compileGeneratedJava (*)

(*) - details omitted (listed previously)
```

I am not understanding why adding the generated.output dependency on the compilation of the main source set is causing xjcGenerate to depend on the generatedClasses task.  If I remove the dependency, xjcGenerate is listed as having no dependencies (as seen using the task-tree plugin) which is what I would expect but obviously the xjc generated files are now no longer included while compiling main, which I need.  Is this a bug in the plugin or am I doing something wrong?

I have attached a small but complete project that demonstrates this problem.
[xjc-sample.zip](https://github.com/unbroken-dome/gradle-xjc-plugin/files/2589573/xjc-sample.zip)

**Comment by aorze** [comment]
I believe the dependency is being added by setting the catalogResolutionClasspath in XjcPlugin.groovy.  If I force this value to be empty instead of the default, I no longer get the circular dependency.

**Comment by tkrullmann** [comment]
Yes, that seems to be what happens. The `catalogResolutionClasspath` is set to the `compileClasspath` by default, and it's marked as `@InputFiles` so it leads to a task dependency from `xjcGenerate` on whatever builds the files in `compileClasspath`.
Since there is a dependency from `compile` on the generated output, `xjcGenerate` now also contributes to the `compileClasspath`, which leads to the circular task dependency.

Trying to come up with a solution!

**Comment by tkrullmann** [comment]
Should be fixed in 1.4.1, please try
